v2/index_gen.go

```diff
@@ -57,11 +62,13 @@ func GenerateIndex(v1r io.Reader) (index.Index, error) {
 			return nil, err
 		}
 
-		// Null padding; treat zero-length frames as an EOF.
-		// They don't contain a CID nor block, so they're not useful.
-		// TODO: Amend the CARv1 spec to explicitly allow this.
+		// Null padding; by default it's an error.
 		if frameLen == 0 {
-			break
+			if o.V1NullPadding {
+				break
+			} else {
+				return nil, fmt.Errorf("carv1 null padding not allowed by default; see AllowV1NullPadding")
+			}
```

Comment:
It's four lines and I seem to understand one or two of these are just copy-pastes from GenerateIndex, so I'd rather see us deduplicate the entire ~30 lines instead of just these 4 :)

Reply:
Sounds good